src/App.ts

```diff
@@ -239,27 +249,38 @@ export default class App {
     });
 
     this.listeners.push(
-      [onlyEvents, matchEventType('message'), ...messageMiddleware] as Middleware<AnyMiddlewareArgs>[],
+      [onlyEvents, matchEventType('message'), ...messageMiddleware] as Middleware<AnyMiddlewareArgs, Context>[],
     );
   }
 
   // NOTE: this is what's called a convenience generic, so that types flow more easily without casting.
   // https://basarat.gitbooks.io/typescript/docs/types/generics.html#design-pattern-convenience-generic
-  public action<ActionType extends SlackAction = SlackAction>(
+  public action<
+    CurrentContext extends Context = StringIndexedContext,
+    ActionType extends SlackAction = SlackAction
```

Comment:
Adding a new type parameter like this results in a breaking change for a portion of users who explicitly give the ActionType here. It's easy to detect as the TypeScript transpiration will fail, though.